src/main/java/pages/BasePage.java

```diff
@@ -1,15 +1,39 @@
 package pages;
 
+import elements.CustomButton;
 import elements.CustomElementDecorator;
+import io.qameta.allure.Step;
+import org.apache.log4j.Logger;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.PageFactory;
 import webdriver.DriverManager;
 
 public class BasePage {
+
+    private static final Logger logger = Logger.getLogger("Base page logger");
+
     protected WebDriver driver;
 
     public BasePage() {
         driver = DriverManager.getDriver();
         PageFactory.initElements(new CustomElementDecorator(this.driver), this);
     }
+
+    @Step("Get page url")
+    public String getPageUrl() {
+        logger.debug("Successfully retrieved page url");
+        return driver.getCurrentUrl();
+    }
+
+    @Step("Refresh page")
+    public HomePage refreshPage() {
```

Comment:
What if you refresh registrationPage? this methos will return homePage, is that right? 
In BasePage there should be methods which can be related to ALL pages.
Make this method REUSABLE for other pages